### DISCUSSION TITLE
How can i fix UndeterminedValue SP

### DISCUSSION THREAD

**Discussion by juuun0** [Vector35/binaryninja-api#5869] [Q&A] [answered]
While i analyze some binary(Aarch64, ELF) i encountered 'Unresolved Stack Pointer Value' like below image:

![Screenshot 2024-08-21 at 13 52 52](https://github.com/user-attachments/assets/73a51d25-28c5-4639-953c-15b1947a0ad5)

For fix this, i found some videos and it tolds me change to LLIL view and put your cursor on `call()` instruction after then right click -> *set stack adjustment* but it isn't work for me because `call()` instruction is not there.

The logic is not complicated to me but i don't know how can i fix this error to correct SP.

1. Store sp-0x9000 at x12 -- sp is not changed here
2. Store sp-0x1000 at sp -- sp is decreased by 0x1000
3. Compare x12 and sp -- if it isn't same, jump to 2

2 and 3 is repeated until x12 and sp has same value so final sp is indicate `-0x9000` from the base in my think.

Let me know is my analysis wrong or any option for fix this error.

thanks.

**Comment by plafosse** [comment, marked as answer]
Two other options are to set the variable value after the loop or to patch the program so as to remove the loop.

**Reply by juuun0** [reply]
I found some menu 'Set User Variable Value' but it doesn't supports on Low Level IL 😢 Let me try your suggested options.
